branch_name_verification.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+import robot.basestation.app as app
+from robot.basestation.app import run_shell
+import re
+
+excluded_branches = {'master', 'develop', 'staging', 'test'}
+
+output, error = run_shell("git", "symbolic-ref --short HEAD", False)
+branch_name = output.decode("utf8")[0:-1]
+
+def is_excluded_branch(branch_name, excluded_branches):
+    """
+    check if branch name is in list of excluded branches
+    """
+    return branch_name in excluded_branches
+
+def has_upper_case_letters(trim_issue_num):
+    """
+    check if trimed branch name only contains lower case characters
+    """
+    return not (trim_issue_num.islower())
+
+def has_hyphen_seperator(trim_issue_num):
+    """
+    check for hyphen seperator between branch name and issue number
+    """
+    return trim_issue_num.endswith('-')
+
+def get_issue_num(branch_name):
+    """
+    Gets issue number from branch named
+    """
+    issue_num = re.search(r'\d+$', branch_name)
+    if issue_num is not None:
+        issue_num = issue_num.group(0)
+    return issue_num
+
+def has_issue_num(branch_name):
+    """
+    check if there is an issue number
+    """
+    return not get_issue_num(branch_name) is None
+
+def has_invalid_symbols(branch_name):
+    """
+    check for invalid invalid_symbols
+    """
+    invalid_symbols = {"/", "'", '!', '?', '@', '$', '%', ',', '.', '_', '+', '£', '=', '¬', '<', '>'}
+    return (bool(set(branch_name) & invalid_symbols)) or ('\\' in branch_name)
+
+def branch_name_error(branch_name, excluded_branches):
```

Comment:
I would not pass or check if the branch should be included in this function. It shouldn't be this functions responsibility to do both checking if it has an error and checking if it was excluded.

Reply:
I have a suggestion for what to do in another comment

Reply:
Moved responsibility to where `__name__ == "__main__"` takes place